### PR TITLE
feat(docs) - add cross-examples for rich components in docs, add mention support

### PR DESCRIPTION
**Pull request by Antreesy**
### ☑️ Resolves
* Extend docs (follow-ups from https://github.com/nextcloud-libraries/nextcloud-vue/pull/4485):
  * NcRichContenteditable - render mentions in NcRichText
  * NcRichText - add example with NcRichContenteditable usage

Commits were separated from #4486 
### 🖼️ Screenshots

🏚️ Before | 🏡 After
---|---
 "-" | NcRichContenteditable docs
image | image
 "-" | NcRichText docs
image | image

### 🏁 Checklist

- [x] ⛑️ Tests are included or are not applicable
- [x] 📘 Component documentation has been extended, updated or is not applicable

### CODE REVIEW

File: src/components/NcRichContenteditable/NcRichContenteditable.vue [hunk 1]
```diff
@@ -55,84 +57,48 @@ This component displays contenteditable div with automated `@` [at] autocompleti
...
+			//       title: 'Test02',
+			//       source: 'users',
+			//       status: {
+			//     clearAt: null,
+			//         icon: '🎡',
+			//         message: 'Visiting London',
+			//         status: 'away',
+			//   },
+			//   subline: 'Visiting London',
+			// },
```
Comment (ShGKme) [inline src/components/NcRichContenteditable/NcRichContenteditable.vue:86]: Supposed to be removed?
```suggestion
```

File: src/components/NcRichContenteditable/NcRichContenteditable.vue [hunk 2]
```diff
@@ -55,84 +57,48 @@ This component displays contenteditable div with automated `@` [at] autocompleti
...
+			userMentions,
+			// Test02: {
+			//   component: 'NcUserBubble',
+			//   props: {
+			//     displayName: 'Test02',
+			//   },
+			// },
```
Comment (ShGKme) [inline src/components/NcRichContenteditable/NcRichContenteditable.vue:93]:
```suggestion
```

File: src/components/NcRichContenteditable/NcRichContenteditable.vue (hunk 1 above)
Comment (Antreesy) [reply src/components/NcRichContenteditable/NcRichContenteditable.vue:86]: `userData` is referenced in the JSDOC description of method below:

https://github.com/nextcloud-libraries/nextcloud-vue/blob/c338863ba242294e68e442980d2974c57a315c38/src/components/NcRichContenteditable/NcRichContenteditable.vue#L141

Now as it's moved to the separate file, and is no longer visible in the styleguide code overview, I've decided to keep an example in docs for developers

File: src/components/NcRichContenteditable/NcRichContenteditable.vue (hunk 2 above)
Comment (Antreesy) [reply src/components/NcRichContenteditable/NcRichContenteditable.vue:93]: Same here, should be treated equally

File: src/components/NcRichText/NcRichText.vue [hunk 3]
```diff
@@ -60,10 +70,167 @@ Some examples for markdown syntax: **bold text** *italic text* \`inline code\``,
...
+	<div>
+		<NcRichContenteditable :value.sync="message"
+			:auto-complete="autoComplete"
+			:maxlength="100"
+			:user-data="userData"
+			placeholder="Try mentioning user @Test01 or inserting emoji :smile"
+			@submit="onSubmit" />
+
+		<NcCheckboxRadioSwitch :checked.sync="autolink" type="checkbox">Autolink</NcCheckboxRadioSwitch>
+		<NcCheckboxRadioSwitch :checked.sync="useMarkdown" type="checkbox">Use Markdown</NcCheckboxRadioSwitch>
```
Comment (DorraJaouad) [inline src/components/NcRichText/NcRichText.vue:97]: Why are you adding the whole code of `NcRichContenteditable` here ? wouldn't the reference to component ( "See ...") be sufficient ?

Comment (Antreesy) [reply src/components/NcRichText/NcRichText.vue:97]: That was the point from my side: to show in both parts of documentation, what these components can be used together - same as we're using `NcAvatar`, `NcModal` or `NcCheckboxRadioSwitch` in other component docs

**Review by ShGKme — APPROVED** [review]

**Pull request merged**